packages/server/src/automations/automationUtils.ts

```diff
@@ -128,55 +135,56 @@ export async function sendAutomationAttachmentsToStorage(
 
 async function generateAttachmentRow(attachment: AutomationAttachment) {
   const prodAppId = context.getProdAppId()
-  try {
-    let size: number | null | undefined
-    let s3Key: string
-    const client = objectStore.ObjectStore(objectStore.ObjectStoreBuckets.APPS)
 
+  async function uploadToS3(
+    extension: string,
+    content: objectStore.StreamTypes
+  ) {
+    const fileName = `${uuid.v4()}.${extension}`
+    const s3Key = `${prodAppId}/attachments/${fileName}`
+
+    await objectStore.streamUpload({
+      bucket: objectStore.ObjectStoreBuckets.APPS,
+      stream: content,
+      filename: s3Key,
+    })
+
+    return s3Key
+  }
+
+  async function getSize(s3Key: string) {
+    return (
+      (
+        await objectStore.getObjectMetadata(
+          objectStore.ObjectStoreBuckets.APPS,
+          s3Key
+        )
+      ).ContentLength || 0
```

Comment:
Is 0 a correct default here? In what scenarios would `ContentLength` by falsey?

Reply:
Honestly, I'm not sure. I guess the `S3.HeadObjectOutput` output can be undefined but I don't think we want to fail it if it can't get the size for whatever reason? What do you think?